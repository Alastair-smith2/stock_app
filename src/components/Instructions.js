import React from "react";
import { View, Text, StyleSheet, Platform } from "react-native";

const Instructions = ({ instructionContainer, label }) => {
  return (
    <View style={instructionContainer}>
      <Text style={styles.instruction}>{label}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  instruction: {
    fontSize: 18,
    textAlign: "center",
    fontFamily: "Futura",
    ...Platform.select({
      ios: {
        color: "black"
      },
      android: {
        color: "#212121"
      }
    })
  }
});

export default Instructions;
