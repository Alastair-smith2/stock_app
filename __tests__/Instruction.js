import React from "react";
import { shallow } from "enzyme";

import Instructions from "../src/components/Instructions.js";

describe("The Instructions component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Instructions />);
  });
  describe("It should render", () => {
    it("Should render", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
