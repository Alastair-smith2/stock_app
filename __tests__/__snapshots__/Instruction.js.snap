// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Instructions component It should render Should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Instructions />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontFamily": "Futura",
            "fontSize": 18,
            "textAlign": "center",
          }
        }
      />,
      "style": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "accessible": true,
        "allowFontScaling": true,
        "children": undefined,
        "ellipsizeMode": "tail",
        "style": Object {
          "fontFamily": "Futura",
          "fontSize": 18,
          "textAlign": "center",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontFamily": "Futura",
              "fontSize": 18,
              "textAlign": "center",
            }
          }
        />,
        "style": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": undefined,
          "ellipsizeMode": "tail",
          "style": Object {
            "fontFamily": "Futura",
            "fontSize": 18,
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
